src/erasurecode_helpers.c

```diff
@@ -208,13 +208,44 @@ int get_aligned_data_size(ec_backend_t instance, int data_len)
 
 /* ==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~== */
 
-char *get_data_ptr_from_fragment(char *buf)
+char *_get_data_ptr_from_fragment(char *buf)
 {
-    buf += sizeof(fragment_header_t);
+    fragment_header_t *header = (fragment_header_t *) buf;
+
+    assert(NULL != header);
+    if (header->magic != LIBERASURECODE_FRAG_HEADER_MAGIC) {
+        log_error("Invalid fragment header (size check)!");
+        return NULL;
+    }
+
+    buf += sizeof(fragment_header_t) + header->meta.frag_backend_metadata_size;
```

Comment:
Is this change applied only for quadiron or also other backends?

Reply:
also for other backends, but in theory this is what should be done if backend_metadata_size is non-zero

Reply:
they want a different changeset, same PR